src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Receiver.java

```diff
@@ -0,0 +1,341 @@
+            {
+                logger.error("Error creating flexfec packet");
+                return;
+            }
+
+            for (Integer protectedSeqNum : fecPacket.getProtectedSequenceNumbers())
+            {
+                logger.debug("Checking if we've received media " +
+                    "packet " + protectedSeqNum);
+                if (!mediaPackets.containsKey(protectedSeqNum))
+                {
+                    logger.debug("We haven't, mark as missing");
+                    numMissing++;
+                    missingSequenceNumber = protectedSeqNum;
+                }
+            }
+            logger.debug("There were " + numMissing + " missing media " +
+                "packets for flexfec " + p.getSequenceNumber());
+            if (numMissing > 1)
+            {
+                missingSequenceNumber = -1;
+            }
+        }
+
+        /**
+         * Initialize the given RawPacket with the RTP header information
+         * and payload from fecPacket
+         * @param fecPacket the FlexFEC packet being used for recovery
+         * @param recoveredPacket the blank RawPacket we're recreating the
+         * recovered packet in
+         * @return true on success, false otherwise
+         */
+        private boolean startPacketRecovery(FlexFec03Packet fecPacket, RawPacket recoveredPacket)
+        {
+            if (fecPacket.getLength() < FlexFec03Packet.FIXED_HEADER_SIZE)
+            {
+                logger.error("Given FlexFEC packet is too small");
+                return false;
+            }
+            if (recoveredPacket.getBuffer().length < RawPacket.FIXED_HEADER_SIZE)
+            {
+                logger.error("Given RawPacket buffer is too small");
+                return false;
+            }
+            // Copy over the recovery RTP header data from the fec packet
+            // (fecPacket contains the RTP header, so we need to copy from it
+            // starting after that)
+            System.arraycopy(fecPacket.getBuffer(), fecPacket.getFlexFecHeaderOffset(),
+                recoveredPacket.getBuffer(), 0, RawPacket.FIXED_HEADER_SIZE);
+
+            // Copy over the recovery rtp payload data from the fec packet
+            System.arraycopy(
+                fecPacket.getBuffer(),
+                fecPacket.getFlexFecHeaderOffset() + fecPacket.getFlexFecHeaderSize(),
+                recoveredPacket.getBuffer(),
+                RawPacket.FIXED_HEADER_SIZE,
+                fecPacket.getFlexFecPayloadLength());
+
+            return true;
+        }
+
+        /**
+         * Xor the RTP headers of source and destination
+         * @param source the packet to xor the header from
+         * @param dest the packet to xor the header into
+         */
+        private void xorHeaders(RawPacket source, RawPacket dest)
+        {
+            // XOR the first 2 bytes of the header: V, P, X, CC, M, PT fields.
+            dest.getBuffer()[0] ^= source.getBuffer()[0];
```

Comment:
The current code creates 'dest' with an offset of 0, but we need to take it into account for 'source'.

Reply:
fixed (i hate offset)